saml/src/main/java/com/linecorp/armeria/saml/HttpRedirectBindingUtil.java

```diff
@@ -0,0 +1,292 @@
+import com.linecorp.armeria.saml.SamlServiceProviderConfigurator.IdentityProviderConfig;
+
+import io.netty.handler.codec.http.HttpHeaderValues;
+import io.netty.handler.codec.http.QueryStringEncoder;
+
+/**
+ * A utility class which supports HTTP-Redirect binding protocol.
+ */
+final class HttpRedirectBindingUtil {
+    private static final Logger logger = LoggerFactory.getLogger(HttpRedirectBindingUtil.class);
+
+    /**
+     * HTTP parameter names which are used for HTTP-Redirect binding protocol.
+     */
+    static final String PARAM_SAML_REQUEST = "SAMLRequest";
+    static final String PARAM_SAML_RESPONSE = "SAMLResponse";
+    static final String PARAM_SIGNATURE = "Signature";
+    static final String PARAM_SIGNATURE_ALGORITHM = "SigAlg";
+    static final String PARAM_RELAY_STATE = "RelayState";
+
+    private static final String DEFAULT_CC =
+            String.join(",", HttpHeaderValues.NO_CACHE, HttpHeaderValues.NO_STORE);
+    private static final String DEFAULT_PRAGMA = HttpHeaderValues.NO_CACHE.toString();
+
+    /**
+     * Returns an {@link HttpHeaders} with the specified {@code location}, a default {@code cache-control}
+     * and a default {@code pragma} headers.
+     */
+    static HttpHeaders headersWithLocation(String location) {
+        return HttpHeaders.of(HttpStatus.FOUND)
+                          .add(HttpHeaderNames.LOCATION, location)
+                          .add(HttpHeaderNames.CACHE_CONTROL, DEFAULT_CC)
+                          .add(HttpHeaderNames.PRAGMA, DEFAULT_PRAGMA);
+    }
+
+    /**
+     * Returns an {@link HttpResponse} with the specified {@code location}, a default {@code cache-control}
+     * and a default {@code pragma} headers.
+     */
+    static HttpResponse responseWithLocation(String location) {
+        return HttpResponse.of(headersWithLocation(location));
+    }
+
+    /**
+     * Returns a redirected URL that the specified SAML message is deflated and base64-encoded.
+     * The URL must contain a signature of the generated query string.
+     */
+    static String toRedirectionUrl(SAMLObject msg,
+                                   String endpointUrl,
+                                   String messageParamName,
+                                   Credential signingCredential,
+                                   String signatureAlgorithm,
+                                   @Nullable String relayState) {
+        requireNonNull(msg, "msg");
+        requireNonNull(endpointUrl, "endpointUrl");
+        requireNonNull(messageParamName, "messageParamName");
+        requireNonNull(signingCredential, "signingCredential");
+        requireNonNull(signatureAlgorithm, "signatureAlgorithm");
+
+        try {
+            final QueryStringEncoder encoder = new QueryStringEncoder(endpointUrl);
+
+            final String samlRequest = deflateAndBase64Encode(msg);
+            encoder.addParam(messageParamName, samlRequest);
+
+            if (relayState != null) {
```

Comment:
How about `checkArgument(relayState == null || relayState.length() <= 80)`. Do you want to ignore this too long case?

Reply:
From SAML specification, it must not exceed 80 bytes in length, but I found that OpenSAML library allows a string which exceeds 80 bytes in length. So I just added a log message here. 

On second thought, it would be better to follow the specification, so let me fix the condition as what you told.

Thanks!